tests/test_non_persistent_deployment.py

```diff
@@ -0,0 +1,231 @@
+# Copyright (c) Microsoft Corporation.
+# SPDX-License-Identifier: Apache-2.0
+
+# DeepSpeed Team
+import pytest
+import os
+import torch
+from types import SimpleNamespace
+from .utils import *  # noqa: F401
+import mii
+''' These fixtures provide a local deployment and ensure teardown '''
+
+
+@pytest.fixture(scope="function")
+def mii_configs(
+    tmpdir: str,
+    dtype: str,
+    tensor_parallel: int,
+    load_with_sys_mem: bool,
+    enable_load_balancing: bool,
+):
+
+    # Create a hostfile for DeepSpeed launcher when load_balancing is enabled
+    hostfile = os.path.join(tmpdir, "hostfile")
+    num_gpu = torch.cuda.device_count()
+    enable_load_balancing = enable_load_balancing
+    if enable_load_balancing:
+        with open(hostfile, "w") as f:
+            f.write(f"localhost slots={num_gpu}")
+
+    return {
+        'dtype': dtype,
+        'tensor_parallel': tensor_parallel,
+        'load_with_sys_mem': load_with_sys_mem,
+        'enable_load_balancing': enable_load_balancing,
+        'replica_num': num_gpu * enable_load_balancing // tensor_parallel,
+        'hostfile': hostfile,
```

Comment:
I think we can leave these out since load balancing isn't supported (but keep `enable_load_balancing` to ensure we get the expected error)